templates/quickstart-hashicorp-vault.template

```diff
@@ -312,364 +578,397 @@ Resources:
+          Fn::Join:
+            - ""
+            - - !Ref QSS3KeyPrefix
+              - "functions/packages/ClusterBootstrap/lambda.zip"
+      Handler: "lambda_function.lambda_handler"
+      Environment:
+        Variables:
+          LeaderElectedSSM: !Ref VaultLeaderElectedSSM
+          LeaderSSM: !Ref VaultLeaderElectedSSM
+          ClusterMembersSSM: !Ref VaultClusterMembersSSM
+          AutoScalingGroup: !Sub "VaultServerAutoScalingGroup-${AWS::StackName}"
+      Runtime: python3.6
+      Timeout: 300
+      Role: !GetAtt ClusterBootstrapLambdaExecutionRole.Arn
+  ClusterBootstrapLambdaExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - lambda.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: "/"
+      Policies:
+      - PolicyName: root
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Resource:
+              - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*'
+            Action:
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
+              - logs:PutLogEvents  
+          # TODO: Restrict permissions to just the SSM Parameters we create
+          - Effect: Allow
+            Resource:
+              - '*'
+            Action:
+              - ssm:*
+          # TODO: Restrict to only our AutoScaling group
+          - Effect: Allow
+            Resource:
+              - '*'
+            Action:
+              - autoscaling:DescribeAutoScalingGroups
+  VaultClientLaunchConfiguration:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Properties:
+      LaunchConfigurationName: !Sub "VaultClientLC-${AWS::StackName}"
       InstanceType:
         Ref: VaultInstanceType
       KeyName:
-        Ref: KeyPair
-      NetworkInterfaces:
-      - DeleteOnTermination: true
-        DeviceIndex: '0'
-        SubnetId:
-          Ref: PrivateSubnet2ID
-        GroupSet:
-        - Ref: VaultSecGroup
+        Ref: KeyPairName
+      SecurityGroups:
+        - !Ref VaultClientSecurityGroup
       ImageId:
         Fn::FindInMap:
         - AWSAMIRegionMap
-        - Ref: AWS::Region
-        - US1604HVM
+        - !Ref 'AWS::Region'
+        - !FindInMap
+          - LinuxAMINameMap
+          - !Ref VaultAMIOS
+          - Code
       BlockDeviceMappings:
       - DeviceName: "/dev/sda1"
         Ebs:
           VolumeSize: 20
           VolumeType: gp2
+      IamInstanceProfile: 
+        Ref: "VaultClientInstanceProfile" 
       UserData:
-        Fn::Base64:
-          Fn::Join:
-          - ''
-          - - "#!/bin/bash -x\n"
-            - 'apt-get -y update
+        Fn::Base64: !Sub
+          - |
+            #!/bin/bash -x
+            
+            # Install git
+            apt-get update
+            apt-get install -y git
```

Comment:
Can you also install `jq` package?

Reply:
I have installed JQ at an earlier point. I will try and consolidate package installations to a single point to make this less confusing.